AudibleApi/Api [partial].cs

```diff
@@ -35,19 +38,35 @@ public Task<HttpResponseMessage> AdHocAuthenticatedGetAsync(string requestUri, I
 
 		public async Task<HttpResponseMessage> AdHocAuthenticatedRequestAsync(string requestUri, HttpMethod method, IHttpClientActions client, JObject requestBody = null)
 		{
-			if (requestUri is null)
-				throw new ArgumentNullException(nameof(requestUri));
-			if (string.IsNullOrWhiteSpace(requestUri))
-				throw new ArgumentException($"{nameof(requestUri)} may not be blank");
-			if (method is null)
-				throw new ArgumentNullException(nameof(method));
-			if (requestBody is null && (method.Method == HttpMethod.Post.Method || method.Method == HttpMethod.Put.Method))
-				throw new ArgumentNullException(nameof(requestBody), $"Must provide request body content when using the {method.Method} {nameof(HttpMethod)}");
-
-			var request = new HttpRequestMessage(method, requestUri);
-
-			if (method.Method == HttpMethod.Post.Method || method.Method == HttpMethod.Put.Method)
+			ArgumentValidator.EnsureNotNullOrWhiteSpace(requestUri, nameof(requestUri));
+			ArgumentValidator.EnsureNotNull(method, nameof(method));
+			ArgumentValidator.EnsureNotNull(client, nameof(client));
+
+			using var request = new HttpRequestMessage(method, requestUri);
+
+			if (method.In(HttpMethod.Post, HttpMethod.Put))
+			{
+				ArgumentValidator.EnsureNotNull(requestBody, nameof(requestBody));
 				request.AddContent(requestBody);
+			}
+
+			request.SignRequest(
+					_identityMaintainer.SystemDateTime.UtcNow,
+					await _identityMaintainer.GetAdpTokenAsync(),
+					await _identityMaintainer.GetPrivateKeyAsync());
+
+			return await SendClientRequest(client, request);
+		}
+
+		public async Task<HttpResponseMessage> AdHocAuthenticatedXmlPostAsync(string requestUri, IHttpClientActions client, XElement requestBody)
+		{
+			ArgumentValidator.EnsureNotNull(requestUri, nameof(requestUri));
+			ArgumentValidator.EnsureNotNull(client, nameof(client));
+			ArgumentValidator.EnsureNotNull(requestBody, nameof(requestBody));
+
+			using var request = new HttpRequestMessage(HttpMethod.Post, requestUri);
+
+			request.AddContent(new StringContent(requestBody.ToString(SaveOptions.DisableFormatting), Encoding.UTF8, "application/xml"));
```

Comment:
You may want to add this as an extension in `Dinah.Core.Net.Http.SystemNetHttpExtensions`

Reply:
You're welcome to do a PR. Or just reply here with exactly how you want me to copy/paste and I'll do it

Reply:
Done. That PR will allow us to keep one generalized `AdHocAuthenticatedRequestAsync` instead of making a new method for different data types.